### PR TITLE
UHF-9158: language switcher titles visible for unpublished translations

### PR DESCRIPTION
**Pull request by rpnykanen**
# [UHF-9158](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9158)

As anonymous user, language switcher would hide a language's title if the node has translation but translation is unpublished.

## What was done
Added language switcher alter which changes the url to another one and sets it disabled. 
- This way the language title will be visible. 
- A disabled switcher link is rendered as span and has no url mentioned in it.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-9158`
* Run `make drush-updb drush-cr`

## How to test
Should be preferably tested with multiple sites (kasko requires drush uli --uid=19 to test group content)
**`Etusivu` for other languages and `Kasko` for group content should be preferably tested.**

- Login as admin, open incognito window as well to test both unauthenticated and logged in user.
- Select an existing content or create new one.
- Make sure you test the content while these rules apply:
  - No translations are enabled
  - One translation is enabled, one is disabled, one does not exist
  - All enabled
- In all cases, language switcher should show all 3 main languages
  - Active language should be always underlined no matter the published state.
  - Published translations should be a link (enabled)
  - Unpublished translation should be text (disabled  span)
- Try publishing and unpublishing content to see that the language switcher updates accordingly

Things to test across all sites:
- User page (entity that is not translatable)
- TPR-pages
- Nodes
- Kasko group content
- Other languages (etusivu)
- Anything else that comes in mind

* [x] Check that this feature works
* [x] Check that code follows our standards

[UHF-9158]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9158?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helfi_platform_config.module [hunk 1]
```diff
@@ -421,3 +421,39 @@ function helfi_platform_config_preprocess_react_and_share(&$variables) : void {
     $variables['privacy_policy_url'] = helfi_eu_cookie_compliance_get_privacy_policy_url();
   }
 }
+
+/**
+ * Implements hook_language_switch_links_alter().
+ *
+ * #UHF-9158 main languages must be always visible on language switcher.
+ */
+function helfi_platform_config_language_switch_links_alter(array &$links) {
```
Comment (tuutti) [inline helfi_platform_config.module:430]: Missing return type.

File: helfi_platform_config.module [hunk 2]
```diff
@@ -421,3 +421,39 @@ function helfi_platform_config_preprocess_react_and_share(&$variables) : void {
...
+  $params = \Drupal::routeMatch()->getParameters()->all();
+
+  $entity = NULL;
+  foreach ($params as $param) {
+    if ($param instanceof ContentEntityInterface) {
+      $entity = $param;
+      break;
+    }
+  }
+  if (!$entity) {
```
Comment (tuutti) [inline helfi_platform_config.module:440]: $entity->isTranslatable() check can be moved here

**Review by tuutti — CHANGES_REQUESTED** [review]

File: tests/src/Functional/LanguageSwitcherAlterTest.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    $this->node->getTranslation('sv')
+      ->set('status', 0)
+      ->save();
+  }
+
+  /**
+   * Tests that languages are visible in language switcher.
+   */
+  public function testLanguageSwitcher() : void {
+    $this->drupalLogout();
```
Comment (tuutti) [inline tests/src/Functional/LanguageSwitcherAlterTest.php:92]: I don't think this is necessary since we shouldn't be logged-in.

**Review by dire — APPROVED** [review]
Worked nicely!

**Review by tuutti — APPROVED** [review]

**Pull request merged**